Doc/library/typing.rst

```diff
@@ -3173,6 +3233,22 @@ Introspection helpers
 
    .. versionadded:: 3.7.4
 
+.. data:: NoDefault
+
+   A sentinel object used to indicate that a type parameter has no default
+   value. For example:
+
+   .. doctest::
+
+      >>> T = TypeVar("T")
+      >>> T.__default__ is typing.NoDefault
+      True
+      >>> S = TypeVar("S", default=None)
+      >>> S.__default__ is None
+      True
```

Comment:
```suggestion
      >>> T.has_default()
      False
      >>> T.__default__ is typing.NoDefault
      True
      >>> S = TypeVar("S", default=None)
      >>> S.has_default()
      True
      >>> S.__default__ is None
      True
```

Reply:
I thought of that but it felt out of place in the docs for `NoDefault`, since the cases you added don't use `NoDefault` at all.

Reply:
Fair enough -- I thought it was quite nice to see together in one example how the two concepts interrelate, but I definitely don't feel strongly!